start.js

```diff
@@ -11,10 +11,84 @@ https://github.com/GPII/prefsEditors/LICENSE.txt
 // Declare dependencies
 /* global require, __dirname */
 
-var util = require("util"),
-    connect = require("kettle/node_modules/express/node_modules/connect"),
-    port = 8888;
+var net = require('net'); 
+var async = require('async');
+var Socket = net.Socket; 
+var kettlePath = process.argv[2];
+var host = process.argv[3];
+var startPort = process.argv[4];
+var endPort = process.argv[5];
 
-connect.createServer(connect["static"](__dirname)).listen(port);
-util.puts("Preferences Management Tool server running...");
-util.puts("Visit http://localhost:" + port + "/demos/prefsEditor/index.html");
+var endingCondition = function(error, port) {
+    if (error) 
+        console.log(error);
+    else
+        startServer(port);
+}
+
+findPort(endingCondition);
+
+function startServer(port) {
+    var connect = require(kettlePath);
+    var browser = require('openurl'); // You should install openurl module. The command is: npm install openurl
+
+    connect.createServer(connect["static"](__dirname)).listen(port);
+    console.log("Preferences Management Tool server running...");
+    console.log("Visit http://"+ host + ":" + port + "/demos/prefsEditor/index.html");
+    browser.open("http://"+ host + ":" + port + "/demos/prefsEditor/index.html");
+}
+
+function findPort (callback) {
```

Comment:
This workflow is unhelpful. We can't have a service running on a port which can't be predicted by the sysadmin and integrator. Please can you remove it and replace it with a simple system which accepts a port from an environment variable.